source/agora/consensus/data/Transaction.d

```diff
@@ -247,3 +274,45 @@ public bool getSumOutput (const Transaction tx, ref Amount acc)
             return false;
     return true;
 }
+
+unittest
+{
+    import vibe.data.json;
+    Transaction old_tx = Transaction(
+        TxType.Data,
+        [Input(Hash.init, 0)],
+        [Output.init],
+        [1,2,3]
+    );
+    auto json_str = old_tx.serializeToJsonString();
+
+    Transaction new_tx = deserializeJson!Transaction(json_str);
+    assert(new_tx.data.length == old_tx.data.length);
+    assert(new_tx.data == old_tx.data);
+}
+
+/*******************************************************************************
+
+    Caculates the fee of transaction data to store
+
+    Params:
+        data_size = The size of the data
+        factor = The factor to calculate for the fee of transaction data
+
+    Return:
+        Fees payable to data storage
+
+*******************************************************************************/
+
+public Amount clculateDataFee(ulong data_size, uint factor) @safe nothrow
```

Comment:
The formula is as follows:

I only dealt with fees for data storage except for normal transaction fees.

<img width="233" alt="" src="https://user-images.githubusercontent.com/11639939/97933741-f95ff180-1db6-11eb-898e-40b1137f349e.png">

<img width="658" alt="" src="https://user-images.githubusercontent.com/11639939/97933760-041a8680-1db7-11eb-84a3-703ef4e4b7ae.png">

<img width="417" alt="" src="https://user-images.githubusercontent.com/11639939/97933794-14cafc80-1db7-11eb-9272-89e0ceb45375.png">

If we adjust the factor, we can change the fee appropriately.
I added factor to ConsensusParams.